internal/fleet/agent_policy/update.go

```diff
@@ -36,14 +38,39 @@ func (r *agentPolicyResource) Update(ctx context.Context, req resource.UpdateReq
 	}
 
 	policyID := planModel.PolicyID.ValueString()
-	policy, diags := fleet.UpdateAgentPolicy(ctx, client, policyID, body)
+
+	// Extract space IDs from plan and determine operational space
+	// Using default-space-first model for stable multi-space updates
+	planSpaceIDs := fleetutils.ExtractSpaceIDs(ctx, planModel.SpaceIds)
+	spaceID := fleetutils.GetOperationalSpace(planSpaceIDs)
+
+	// Update using the operational space
+	// The API will handle adding/removing the policy from spaces based on space_ids in body
+	var policy *kbapi.AgentPolicy
+	if spaceID != nil && *spaceID != "" {
+		policy, diags = fleet.UpdateAgentPolicyInSpace(ctx, client, policyID, *spaceID, body)
+	} else {
+		policy, diags = fleet.UpdateAgentPolicy(ctx, client, policyID, body)
+	}
+
 	resp.Diagnostics.Append(diags...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
+	// Preserve the space_ids order from plan before populating from API
+	// The Kibana API may return space_ids in a different order (sorted),
+	// but we need to maintain the user's configured order to avoid false drift detection
+	originalSpaceIds := planModel.SpaceIds
+
 	planModel.populateFromAPI(ctx, policy)
 
+	// Restore the original space_ids order if appropriate
+	// See ShouldPreserveSpaceIdsOrder documentation for edge case handling
+	if fleetutils.ShouldPreserveSpaceIdsOrder(policy.SpaceIds, originalSpaceIds, planModel.SpaceIds) {
```

Comment:
We can do away with this `ShouldPreserveSpaceIdsOrder` palaver if we switch `space_ids` to be a `SetAttribute` rather than a `ListAttribute`.